pkg/test/framework/components/istio/operator.go

```diff
@@ -464,17 +475,22 @@ func installControlPlaneCluster(i *operatorComponent, cfg Config, cluster resour
 	var istiodAddress net.TCPAddr
 	if !cluster.IsConfig() {
 		configCluster := cluster.Config()
+		if err != nil {
+			return err
+		}
 		istiodAddress, err = i.RemoteDiscoveryAddressFor(configCluster)
 		if err != nil {
 			return err
 		}
 		if err := patchIstiodCustomHost(istiodAddress, cfg, cluster); err != nil {
 			return err
 		}
-		if err := configureDiscoveryForConfigCluster(istiodAddress.IP.String(), cfg, configCluster); err != nil {
+		if err := configureDiscoveryForConfigAndRemoteCluster(istiodAddress.IP.String(), cfg, configCluster); err != nil {
```

Comment:
Isn't discovery only needed for remote clusters? Config clusters should not use discovery.